packages/core/utils/lib/sanitize/index.js

```diff
@@ -37,12 +37,16 @@ const createContentAPISanitizers = () => {
     return pipeAsync(...transforms)(data);
   };
 
-  const sanitizeOuput = (data, schema, { auth } = {}) => {
+  const sanitizeOutput = async (data, schema, { auth } = {}) => {
     if (isArray(data)) {
-      return Promise.all(data.map((entry) => sanitizeOuput(entry, schema, { auth })));
+      const res = new Array(data.length);
+      for (let i = 0; i < data.length; i += 1) {
+        res[i] = await sanitizeOutput(data[i], schema, { auth });
+      }
+      return res;
```

Comment:
Did you tried the mapAsync func to have concurrent promise resolution? 🤔 
Wondering if it would be faster or slower

Reply:
I just tried and it has similar response time. Almost all of the operations are sync, so we don't get much benefit from concurrent asyncronous ops here.

Reply:
Just the opposite, everything (almost) is sync and we get the overhead of doing everything async